packages/ng/icon/icon.component.ts

```diff
@@ -0,0 +1,30 @@
+import { ChangeDetectionStrategy, Component, HostBinding, inject, Input, OnInit } from '@angular/core';
+import { CommonModule, NgClass } from '@angular/common';
+import { LuccaIcon } from '@lucca-front/icons';
+
+@Component({
+	selector: 'lu-icon',
+	standalone: true,
+	imports: [CommonModule],
+	hostDirectives: [NgClass],
+	template: '',
+	styleUrls: ['./icon.component.scss'],
+	changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class IconComponent implements OnInit {
+	#ngClass = inject(NgClass);
+
+	@HostBinding('attr.aria-hidden')
+	readonly ariaHidden = true;
+
+	@Input({ required: true })
+	set icon(icon: LuccaIcon) {
+		this.#ngClass.ngClass = {
+			[`icon-${icon}`]: true,
+		};
+	}
```

Comment:
You're right, I didn't think of this kind of use case, I should just handle everything in ngClass.

Reply:
Actually that's kind of a headache, because if we have two sources updating the same ngClass, it might not be possible to make them behave properly.

Imagine you have something like `<lu-icon icon="xxx" class="foo bar" ngClass="computedClasses$ | async"/>`, then we'd have an issue with two ngClass sources overrriding each other, and since ngClass doesn't have a way to merge these, I feel like this means that we can't really use ngClass in this case. @GuillaumeNury this is an interesting thing to consider regarding the `inject(ngClass)` approach that we might have overlooked 😄

Reply:
There is indeed a problem with this approach, I opened an [issue on Angular](https://github.com/angular/angular/issues/52072).

In the meantime, we can "copy" NgClass. But it is less elegant:

```ts
@Directive({
	selector: '[ngClazz]',
	standalone: true,
})
class NgClazz extends NgClass {}
```

Reply:
Well it no longer is a concern anyways because turns out we want an inner structure to be able to display an alt text